client/src/pages/Signup.jsx

```diff
@@ -1,258 +1,37 @@
-import {
-  CssBaseline,
-  Box,
-  Grid,
-  Paper,
-  TextField,
-  Typography,
-} from "@material-ui/core";
-import { makeStyles } from "@material-ui/core/styles";
 import { useHistory } from "react-router-dom";
-import { Formik } from "formik";
-import * as Yup from "yup";
-import {
-  IndexPosterComponent,
-  IndexNavHeaderComponent,
-  IndexSnackbarComponent,
-  IndexTitleComponent,
-  IndexSubmitBtnComponent,
-} from "../components/Components";
+import { SignupComponent } from "../components/Components";
+import { useDispatch } from "react-redux";
+import { openSnackbar } from "../store/actionCreator";
 
-const useStyles = makeStyles((theme) => ({
-  root: {
-    minHeight: "100vh",
-    "& .MuiInput-underline:before": {
-      borderBottom: "1.2px solid rgba(0, 0, 0, 0.2)",
-    },
-  },
-  heroText: {
-    fontSize: 26,
-    fontFamily: "'Open Sans'",
-    textAlign: "center",
-    color: "white",
-    marginTop: 30,
-    maxWidth: 300,
-  },
-  buttonHeader: {
-    display: "flex",
-    alignItems: "flex-start",
-    justifyContent: "space-between",
-    flexDirection: "column",
-    bgcolor: "background.paper",
-    minHeight: "100vh",
-    paddingTop: 23,
-  },
-  image: {
-    backgroundImage: "url(./images/bg-img.png)",
-    backgroundRepeat: "no-repeat",
-    backgroundSize: "cover",
-    backgroundPosition: "center",
-  },
-  box: {
-    padding: 24,
-    display: "flex",
-    justifyContent: "center",
-    alignItems: "center",
-    minHeight: "100vh",
-    flexDirection: "column",
-    maxWidth: 900,
-    margin: "auto",
-  },
-  form: {
-    width: "100%", // Fix IE 11 issue.
-    marginTop: theme.spacing(1),
-  },
-  label: { fontSize: 19, color: "rgb(0,0,0,0.4)", paddingLeft: "5px" },
-  inputs: {
-    marginTop: ".8rem",
-    height: "2rem",
-    padding: "5px",
-  },
-}));
-
-function useRegister(setOpen, setSignupMsg) {
-  const history = useHistory();
+export default function Register() {
+  const dispatch = useDispatch();
 
-  const login = async (username, email, password) => {
-    try {
-      const res = await fetch(`/auth/signup`, {
+  const useRegister = (username, email, password) => {
+    const history = useHistory();
+    const register = async (username, email, password) => {
+      fetch(`/auth/signup`, {
         method: "POST",
-        headers: { "Content-Type": "application/json" },
+        headers: {
+          "Content-Type": "application/json",
+        },
         body: JSON.stringify({
           email: email,
           username: username,
           password: password,
         }),
-      }).then((res) => res.json());
-      if (res.data.signedup) {
-        history.push("/login");
-      } else {
-        setSignupMsg(res.data.msg);
-        setOpen(true);
-      }
-    } catch (error) {
-      history.push("/400");
-    }
+      })
+        .then((res) => res.json())
+        .then((res) => {
+          if (res.data.signedup) {
+            history.push("/login");
+          } else {
```

Comment:
Might want to push to the dashboard rather than the login page here.

Reply:
Fixed it.